### ISSUE TITLE
res `undefined` case not handled

### ISSUE THREAD

**Issue by meienberger** [TheEdoRan/next-safe-action#28]
## Purpose
It could happen that the return value of `clientCaller` is undefined (for example when the next server is not running anymore) this results in an app crash whereas the hook should just enter `onError`
 
![Screenshot 2023-10-12 at 22 09 27](https://github.com/TheEdoRan/next-safe-action/assets/47644445/839e0e73-d435-4661-ae0f-7f3056b52b73)

An easy fix would be to access the `.data` property safely in `getActionStatus`

```typescript
const hasSucceded = typeof res?.data !== "undefined";
const hasErrored =
    typeof res === "undefined" ||
    typeof res.validationError !== "undefined" ||
    typeof res.serverError !== "undefined" ||
    typeof res.fetchError !== "undefined";
```

**Comment by meienberger** [comment]
Might be resolved by https://github.com/TheEdoRan/next-safe-action/pull/26

**Comment by TheEdoRan** [comment]
Thank you for opening this issue! `clientCaller` and the code snippet you provided are from next-safe-action v3, but the new code for handling action status is pretty similar, so I think the same thing happens with version 4 as well.

> for example when the next server is not running anymore

Could you please tell me when this might happen? Isn't the app already dead if Next server is no longer running?

**Comment by meienberger** [comment]
Sure, for example my app is a self-hosted service users can run on their server with docker. In the settings you can restart the app which will stop the containers and restart it. During this time, I keep the frontend loaded, show a spinner and start polling for the endpoint. Once it responds again, I know the app successfully restarted.

**Comment by TheEdoRan** [comment]
Thank you for explaining your use case! As you already said, this will be fixed by merging #26, so when the PR is ready I'll release a new version.
